swarm/runtime/diff_scanner.py

```
@@ -380,15 +339,110 @@ def scan_file_changes_sync(
+    return _process_scan_results(
+        numstat_stdout,
+        status_stdout,
+        include_untracked,
+        include_staged,
+    )
+
+
+async def _run_git_command_async(
+    args: List[str],
+    cwd: Path,
+    timeout: float = 30.0,
```

Ruff (0.15.2)

[warning] Async function definition with a `timeout` parameter

(ASYNC109)